### ISSUE TITLE
How to convert WPF Model3D/GeometryModel3D to Xbim object for later save it to IFC file format?

### ISSUE THREAD

**Issue by renanfaccin** [xBimTeam/XbimEssentials#308]
How to convert WPF Model3D/GeometryModel3D to Xbim object for later save it to IFC file format?

**Comment by andyward** [comment]
There's no _right_ answer to this - it really depends what you want to achieve in the wider use case once the IFC is generated.

IFC supports various different geometry representations.  E.g. CSGs, Extruded Solids, Surface, BREPs, NURBS, to name a few. E.g. See https://standards.buildingsmart.org/IFC/DEV/IFC4_2/FINAL/HTML/link/basic-geometric-shape.htm

... but AFAIK `GeometryModel3D` **only** implements a triangulated mesh intended to be used by GPUs. So basically any `IFC => GeometryModel3D =>IFC` will be a lossy process (unless the IFC model was tessellated). As an example if you took a cylindrical pipe - which may be modeled as an extruded circle along a path, once converted into a tessellated model for it'll be very hard to reconstruct the original extruded representation. 

If this is still what you want to do, and you can use IFC4, saving to a [IfcTriangulatedFaceSet](https://standards.buildingsmart.org/IFC/DEV/IFC4_2/FINAL/HTML/schema/ifcgeometricmodelresource/lexical/ifctriangulatedfaceset.htm) may be the easiest thing to do. Alternately BREP is probably the next best.

Bottom line is that using a tessellated format as the 'master' will be problematic, if you're editing geometry. You likely want to have a parametric 'master' model that you translate to a mesh as needed. The Xbim Geometry engine has those primitives but it's a complex domain to work in.
